drivers/Makefile.dep

```diff
@@ -608,6 +608,23 @@ ifneq (,$(filter w5100,$(USEMODULE)))
   USEMODULE += luid
 endif
 
+ifneq (,$(filter ws281x_%,$(USEMODULE)))
+  USEMODULE += ws281x
+endif
+
+ifneq (,$(filter ws281x,$(USEMODULE)))
+  FEATURES_OPTIONAL += arch_avr8
+  ifeq (,$(filter ws281x_%,$(USEMODULE)))
+    ifneq (,$(filter arch_avr8,$(FEATURES_USED)))
+      USEMODULE += ws281x_atmega
+    endif
+  endif
+  ifneq (,$(filter ws281x_atmega,$(USEMODULE)))
+    FEATURES_REQUIRED += arch_avr8
+  endif
```

Comment:
Dito.

I think a user should just always `USEMODULE += ws281x` and not select the backend manually.

Reply:
But there is no way to prevent this. (And if two backends are available for a specific scenario, there might be a reason to ask explictly for one). A proper safeguard might prevent strange bug reports. (E.g. some users with an Arduino Due might think it is an AVR board and ask explicitly for the atmega backend. An explicit error message rather than some strange syntax errors on the inline assembly should be preferable.)